core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt

```diff
@@ -62,15 +35,51 @@ class Primitive<T : ValidType> : TypeExpression<T> {
     }
 }
 
-fun String.toStringType(): Primitive<StringType> = Primitive(this)
+class NumberPrimitive(private val value: Number) : Primitive<NumberType>() {
+    override fun toDopeQuery() = DopeQuery(
+        queryString = "$value",
+        parameters = emptyMap(),
+    )
+}
+
+class StringPrimitive(private val value: String) : Primitive<StringType>() {
+    override fun toDopeQuery() = DopeQuery(
+        queryString = "\"$value\"",
+        parameters = emptyMap(),
+    )
+}
+
+class BooleanPrimitive(private val value: Boolean) : Primitive<BooleanType>() {
+    override fun toDopeQuery() = DopeQuery(
+        queryString = when (value) {
+            true -> TRUE.toDopeQuery().queryString
+            false -> FALSE.toDopeQuery().queryString
+        },
+        parameters = emptyMap(),
+    )
+}
+
+class ArrayPrimitive<T : ValidType>(private val collection: Collection<TypeExpression<out T>>) : Primitive<ArrayType<T>>() {
+    override fun toDopeQuery(): DopeQuery {
+        val dopeQueries = collection.map { it.toDopeQuery() }
+        return DopeQuery(
+            queryString = dopeQueries.joinToString(separator = ", ", prefix = "[", postfix = "]") { it.queryString },
+            parameters = dopeQueries.fold(emptyMap()) { parameters, dopeQueryElement ->
+                parameters + dopeQueryElement.parameters
+            },
+        )
+    }
+}
+
+fun String.toStringType(): StringPrimitive = StringPrimitive(this)
 
-fun Number.toNumberType(): Primitive<NumberType> = Primitive(this)
+fun Number.toNumberType(): NumberPrimitive = NumberPrimitive(this)
 
-fun Boolean.toBooleanType(): Primitive<BooleanType> = Primitive(this)
+fun Boolean.toBooleanType(): BooleanPrimitive = BooleanPrimitive(this)
 
-fun <T : ValidType> Collection<TypeExpression<out T>>.toArrayType(): Primitive<ArrayType<T>> = Primitive(this)
+fun <T : ValidType> Collection<TypeExpression<out T>>.toArrayType() = ArrayPrimitive(this)
 
-private enum class PrimitiveType {
+enum class PrimitiveType {
```

Comment:
This enum is a bit useless. It's basically only used as a wrapper for a string. Would remove it and use proper data object (using my approach from above):
`data object NULL: Primitive<NullType>(DopeQuery("NULL", emptyMap()))`

Reply:
done